packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -132,3 +110,41 @@ export const SelectedItemCheck = styled( Ariakit.SelectItemCheck )`
 	margin-inline-start: ${ ITEM_PADDING };
 	font-size: 24px; // Size of checkmark icon
 `;
+
+const truncateStyles = css`
+	overflow: hidden;
+	text-overflow: ellipsis;
+	white-space: nowrap;
+`;
+
+export const ItemHintWrapper = styled.div`
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	flex-wrap: wrap;
+	flex: 1;
+`;
+
+export const SelectedExperimentalHintWrapper = styled.div`
+	${ truncateStyles }
+`;
+
+export const SelectedExperimentalHintItem = styled.span`
+	color: ${ COLORS.theme.gray[ 600 ] };
+	margin-inline-start: ${ space( 2 ) };
+`;
+
+export const ItemHintContent = styled.span`
+	padding-inline-end: ${ space( 4 ) };
+`;
+
+export const ItemHint = styled.span`
+	// Necessary to override V1 styles passed via the legacy
+	// '.components-custom-select-control__item-hint' class
+	${ SelectItem } && {
+		color: ${ COLORS.theme.gray[ 600 ] };
+		text-align: initial;
+		line-height: ${ CONFIG.fontLineHeightBase };
```

Comment:
The hint needs to use the default `line-height` value (and not the one of the option item), since it could render multi-line text. When rendered on the same line as the option item text, vertical alignment is still [guaranteed by flexbox](https://github.com/WordPress/gutenberg/pull/62848/files#r1654481023).